test/jdk/java/foreign/StdLibTest.java

```diff
@@ -62,6 +62,16 @@ public class StdLibTest extends NativeTestHelper {
 
     final static Charset nativeCharset = Charset.forName(System.getProperty("native.encoding"));
 
+    static MemorySegment allocateSegmentForString(Arena arena, String str) {
+        try {
+            return arena.allocateFrom(str, nativeCharset);
+        } catch (IllegalArgumentException ex) {
+            // Allocate a memory segment using the native encoded bytes
+            byte[] nativeBytes = str.getBytes(nativeCharset);
+            return arena.allocateFrom(ValueLayout.JAVA_BYTE, Arrays.copyOf(nativeBytes, nativeBytes.length + 1));
+        }
```

Comment:
I don't think using `try` is appropriate here; just use the `catch` block (with a hyphen in "native-encoded" and a period at the end of the sentence):
```java
        // Allocate a memory segment using the native-encoded bytes.
        byte[] nativeBytes = str.getBytes(nativeCharset);
        return arena.allocateFrom(ValueLayout.JAVA_BYTE, Arrays.copyOf(nativeBytes, nativeBytes.length + 1));
```

Reply:
Changed in https://github.com/ibmruntimes/openj9-openjdk-jdk/compare/839d97322d6d32906483aef112ec5657eba9639e..748da917b346468236b8ee444c7581e06ffce316